angular-frontend/src/app/graph/graph-data-renderer.service.ts

```diff
@@ -0,0 +1,112 @@
+import {effect, inject, Injectable, signal} from '@angular/core';
+import {DataSourceService} from './graph-data.service';
+import {ScaleLinear, ScaleTime} from 'd3';
+import {DataFormat} from '../omnai-datasource/omnai-scope-server/live-data.service';
+import {line as d3Line} from 'd3-shape';
+
+
+@Injectable({
+  providedIn: 'root',
+})
+/**
+ * This class exists to render the Data of a Graph to it's svg path representation.
+ * It uses a WebWorker to keep the main javascript thread free, if possible.
+ * If the WebWorker cannot be loaded (which should only happen during testing), it falls back to rendering it on the main javascript thread.
+ */
+export class GraphDataRendererService {
+  //Try to start a WebWorker per new instance.
+  constructor() {
+    try{
+      this.worker = new Worker(new URL('./webworker', import.meta.url));
+
+      this.worker.addEventListener("message", (e) =>{
+        this.lastWorkerRequestDone = true;
+        if (!Array.isArray(e.data)) {
+          console.error("recieved invalid path data from webworker: ", e.data);
+          return;
+        }
+        this.#paths.set(e.data);
+      })
+
+      this.worker.addEventListener("messageerror", (e) =>{
+        this.lastWorkerRequestDone = true;
+        console.error("recieved error from webworker: ", e);
+      })
+
+    } catch (e) {
+      console.error('Could not start WebWorker for GraphRendering: ', e);
+    }
+  }
+
+  /**
+   * Data to render
+   * @private
+   */
+  private readonly graphData = inject(DataSourceService);
```

Comment:
This is somehow a difference instance, than the GraphData being used inside `graph.component.ts`.

I noticed this, because the `graphDimensions` did not update in this instance. `this.graphData.graphDimensions()` is different from logging `this.graphDimensions()` in e.g. the `xScale` computed signal.